feat: add tutorial for Swap&Bridge (#526)
cowprotocol/docs: docs/cow-protocol/tutorials/cow-swap/swap-and-bridge.mdx

---

## 1. Setting up a Swap with Different Network Assets

When using **Swap & Bridge**, you’ll start by selecting the tokens you want to trade — just like a regular swap — but with the option to choose a different destination chain.

### Source Token Selector

Open a source token selector:

<img src={SourceTokenSelectorPlaceImg.src} style={{ maxWidth: 500 }} alt="Open source token selector" />

Select the network and token you want to sell:

<img src={SourceTokenSelectorImg.src} style={{ maxWidth: 500 }} alt="Source token selector" />

### Destination Token Selector

Similarly to Source Token Selector, open the selector, choose a network and token you would like to buy

<img src={DestinationTokenSelectorImg.src} style={{ maxWidth: 500 }} alt="Destination token selector" />

:::tip

The list of available destination networks depends on the source token you’ve selected.

If you choose a destination network different from the source token’s network:
- You will only see tokens supported by the bridge provider for that chain.

If the destination network is the same as the source network:
- You’ll see the regular list of tokens from the token lists.

:::

:::note

> **Tokens must be different.** Bridging the same token between two networks is not yet supported. This feature is under development.

Reviewer: To set expectations, I would not say that. Something like:

```suggestion
**Tokens must be different.** Bridging the same token between two networks is not yet supported. This feature will be available in the future.
```

:::

---

## 2. Creating a Swap & Bridge Trade

Once your tokens and networks are set, you can proceed to create the order.

### Token Approval

If it’s your first time selling this token through CoW Swap, you’ll need to approve it.
This is the standard ERC-20 approval process that allows CoW Protocol to use your tokens for the trade.

Approvals can happen through:

* Regular ERC-20 allowance transactions
* [EIP-6963](https://eips.ethereum.org/EIPS/eip-6963) permit flow (when supported)

> <img src={SignApprovalImg.src} alt="Sign an approval" />

Reviewer: Perfect! I came here to say the same thing https://eips.ethereum.org/EIPS/eip-2612

### Signing Steps

> 1. **Sign the post-hook call** for the bridge provider.

Reviewer: This section could be expanded a bit further.
Explain the order and what is being signed in the first request (the bridge calldata).

Author: Thanks!
Added more details:

<img width="995" height="345" alt="image" src="https://github.com/user-attachments/assets/1ae9ce6f-d643-442e-b9ba-c6890c82a98f" />

* **Bridge cost** → A fee charged by a bridge provider for processing your transfer.
* **Estimated bridging time** → Pulled directly from a bridge provider (varies by a network and a provider).
* Bridge **Min. to receive** → Calculated as: `(Swap min. to receive)` - `(Bridge provider cost)`
 

<img src={QuotePreviewImg.src} style={{ maxWidth: 500 }} alt="Quote preview" />

---

## 5. Understanding the Account Proxy

The **Account Proxy** (also known as **CoW Shed**) is a helper contract that powers advanced features like [**CoW Hooks**](/cow-protocol/reference/core/intents/hooks).

* **CoW Hooks** allow solvers to execute additional Ethereum calls before or after your order is settled.
* Your funds are safe:
If anything goes wrong during execution, your assets will be sent to your personal proxy account. You can withdraw them to your wallet at any time.

>More detailed information can be found in [Account Proxy page](https://swap.cow.fi/#/1/account-proxy/help).

---

## 6. How Bridging Execution Works

1. The bridge step will only start **after** a successful swap.

<img src={SwapSuccessImg.src} style={{ maxWidth: 500 }} alt="Swap is finished" />

> >If your order expires, is canceled, or fails, bridging will not start — your funds will remain in your wallet.

Author: Nice!

2. **Sign the order**:

* Sign an off-chain order (ERC-20 token selling).
* Sign a transaction (Native tokens (e.g., ETH) selling).

<img src={SignPostHookImg.src} alt="Sign a post-hook" />
<img src={SignOrderImg.src} alt="Sign an order" />

:::note

**Intermediate token** — the currency received on the source chain as the result of a swap, which is then deposited to the bridge provider.
It is usually the same currency as on the destination chain, but issued on the source blockchain.

:::

:::note

If a token to sell does not have an approved permit, the process will involve three signing steps, beginning with the permit signature.

:::

---

## 3. Using a Custom Recipient

By default, your connected wallet address will be set as the recipient on the destination chain.
If you want to send funds to another address:

* Enable **Custom recipient** in the settings (same place as in the regular Swap).
* Enter a recipient address.
* **Note:** ENS names are **not** supported here — you must use a full wallet address.

<img src={CustomRecipientImg.src} style={{ maxWidth: 600 }} alt="Custom recipient" />

---

## 4. Previewing Your Quote

> In the Swap form you can see a detailed breakdown of the expected swap and bridge:

Reviewer 2: cross-chain swap.

Author: In that case, it is correct, because we are talking about swap and then bridge details:

<img width="497" height="513" alt="image" src="https://github.com/user-attachments/assets/1c317b06-0dfd-40d4-8df8-37d2f1f2015c" />
